tensorflow/python/keras/optimizer_v2/optimizer_v2.py

```diff
@@ -338,6 +338,12 @@ def my_gradient_transformer(grads_and_vars):
       if k not in allowed_kwargs:
         raise TypeError("Unexpected keyword argument "
                         "passed to optimizer: " + str(k))
+      if k == "lr" and (isinstance(kwargs[k],
+                        (ops.Tensor,
+                        learning_rate_schedule.LearningRateSchedule))
+                        or callable(kwargs[k])):
```

Comment:
If we intend to issue a deprecation warning for `lr` (which is fine), we can issue it unconditionally, without additional checks on the type.

Reply:
I don't know. Let me know if it could be a deprecation warning or not.